### PR TITLE
[PW-7100] Add CardOnFile and UnshceduledCardOnFile to Vault Card

### PR DESCRIPTION
**Pull request by RokPopov**
**Description**
Currently when vault is selected for card payments, we are always sending Subscription (`Requests::buildCardRecurringData`) as the `tokenType`. This should be changed to utilise the `recurring_card_token_type`, which is currently only used when `Adyen Tokenization` is selected in the admin panel.

With this PR, when `Vault` is selected in the admin panel, all `CardOnFile` recurring payments will be CVC-less.

**Tested scenarios**
- saved a token with the incorrect `tokenType` (`UnscheduledCardOnFile` and `Subscription`) for a logged-in user, proceeded to the checkout page and made sure that the stored payment methods are not showing in the payment methods list
- saved a token with the correct `tokenType` (`CardOnFile`) for a logged-in user, proceeded to the checkout page and made sure that the stored payment methods are in the showing payment methods list
- proceeded with placing the order, selecting the stored payment method with `CardOnFile` `tokenType` and made sure the payment was successful

### CODE REVIEW

File: view/frontend/web/template/payment/payment-method-vault.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
```
[the remark this answers is not in the document]
Comment (RokPopov) [reply view/frontend/web/template/payment/payment-method-vault.html:6]: oeps!

File: Model/Ui/TokenUiComponentProvider.php [hunk 2]
```diff
@@ -27,6 +43,22 @@ class TokenUiComponentProvider extends AdyenUiComponentProvider implements Token
      */
     public function getComponentForToken(PaymentTokenInterface $paymentToken): TokenUiComponentInterface
     {
-        return $this->getCardComponentForToken($paymentToken);
+        $tokenType = $this->vaultHelper->getAdyenTokenType($paymentToken);
+        $details = json_decode($paymentToken->getTokenDetails() ?: '{}', true);
+        $details['icon'] = $this->dataHelper->getVariantIcon($details['type']);
+        $createdAt = new \DateTime($paymentToken->getCreatedAt());
+        $details['created'] = $createdAt->format('Y-m-d');
+        $details['displayToken'] = $tokenType === Recurring::CARD_ON_FILE || !isset($tokenType);
```
Comment (Morerice) [inline Model/Ui/TokenUiComponentProvider.php:51]:
```suggestion
        $details['displayToken'] = $tokenType === Recurring::CARD_ON_FILE || is_null($tokenType);
```

File: Plugin/CustomerFilterVaultTokens.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+class CustomerFilterVaultTokens
+{
+    /**
+     * Returns filtered list of payment tokens for current customer session
+     * @param CustomerTokenManagement $customerTokenManagement
+     * @param array $customerSessionTokens
+     * @return CustomerTokenManagement[]
+     */
+    public function afterGetCustomerSessionTokens(CustomerTokenManagement $customerTokenManagement, array $customerSessionTokens)
```
Comment (Morerice) [inline Plugin/CustomerFilterVaultTokens.php:25]:
```suggestion
    public function afterGetCustomerSessionTokens(CustomerTokenManagement $customerTokenManagement, array $customerSessionTokens): CustomerTokenManagement[]
```

File: etc/adminhtml/system/adyen_oneclick.xml [hunk 4]
```diff
@@ -38,16 +38,12 @@
         <field id="recurring_card_token_type" translate="label" type="select" sortOrder="20" showInDefault="1" showInWebsite="1" showInStore="1">
             <label>Token Type</label>
             <tooltip>
-                Choose between CardOnFile (one-click payments) or Subscription (fixed schedule payments)
+                Choose between CardOnFile (one-click CVC-less payments), UnscheduledCardOnFile (non-fixed schedule one-click CVC-less payments) or Subscription (fixed schedule payments)
```
Comment (Morerice) [inline etc/adminhtml/system/adyen_oneclick.xml:41]:
```suggestion
                Choose between CardOnFile (one-click), UnscheduledCardOnFile (non-fixed schedule MIT payments) or Subscription (fixed schedule MIT payments)
```

**Review by Morerice — CHANGES_REQUESTED** [review]

**Review by acampos1916 — APPROVED** [review]

**Review by Morerice — APPROVED** [review]

**Pull request merged**